File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOWatchDialParamModel.kt
//
//  IDOWatchDialParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Set watch face event number

open class IDOWatchDialParamModel(dialId: Int) : IDOBaseModel {

    /// ID of the watch face to be set
    /// Dial id
    /// 0 invalid,currently supports1~4 
    @SerializedName("dial_id")
    var dialId: Int = dialId

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOWatchDialParamModel.swift
//
//  IDOWatchDialParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set watch face event number
@objcMembers
public class IDOWatchDialParamModel: NSObject, IDOBaseModel {
    /// ID of the watch face to be set
    /// Dial id
    /// 0 invalid,currently supports1~4 
    public var dialId: Int
    
    enum CodingKeys: String, CodingKey {
        case dialId = "dial_id"
    }
    
    public init(dialId: Int) {
        self.dialId = dialId
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

